src/components/slidePanel/createSlidePanel.ts

```diff
@@ -0,0 +1,175 @@
+import { DNode, Widget, WidgetProperties, WidgetFactory } from '@dojo/widget-core/interfaces';
+import createWidgetBase from '@dojo/widget-core/createWidgetBase';
+import { v } from '@dojo/widget-core/d';
+
+import * as css from './styles/slidePanel.css';
+import * as animations from '../../styles/animations.css';
+import themeable, { ThemeableMixin } from '@dojo/widget-core/mixins/themeable';
+
+export interface SlidePanelProperties extends WidgetProperties {
+	align?: string;
+	open?: boolean;
+	underlay?: boolean;
+	width?: number;
+	onOpen?(): void;
+	onRequestClose?(): void;
+};
+
+export type SlidePanel = Widget<SlidePanelProperties> & ThemeableMixin & {
+	onSwipeStart?(event: MouseEvent & TouchEvent): void;
+	onSwipeMove?(event: MouseEvent & TouchEvent): void;
+	onSwipeEnd?(event: MouseEvent & TouchEvent): void;
+	afterCreate?(element: HTMLElement): void;
+	onTransitionEnd?(event: TransitionEvent): void;
+};
+
+export interface SlidePanelFactory extends WidgetFactory<SlidePanel, SlidePanelProperties> { };
+
+let content: HTMLElement;
```

Comment:
Do these not have to be stored in a weakmap keyed by the instance? Otherwise having multiple instances will mess with the values.